_posts/2018-12-24-snowplow-snowflake-loader-0.4.0-released/index.md

```diff
@@ -0,0 +1,92 @@
+5. [Getting help](#help)
+
+<!--more-->
+
+<h2 id="crossbatch-dedupe">1. Cross-batch deduplication</h2>
+
+Version 0.4.0 introduces [DynamoDB][dynamodb]-powered cross-batch natural deduplication, which works by extracting two identifier properties from an event (`event_id` and `event_fingerprint`), as well as `etl_tstamp` which identifies a single batch, then storing these properties in a DynamoDB table. Duplicate events with the same `event_id` and `event_fingerprint`, but a different `etl_tstamp`, found in storage are silently dropped from the Snowflake Transformer output.
+
+Cross-batch deduplication can be enabled by creating an additional config file with the following properties:
+
+* `name` - Required human-readable configuration name, e.g. `Snowflake deduplication config`
+* `id` - Required machine-readable configuration id, e.g. UUID
+* `auth` - An object containing information about authentication use to read and write data to DynamoDB. Similar to the `auth` object in the main Snowflake config, this can use a `accessKeyId`/`secretAccessKey` pair or be set to `null`, in which case default credentials will be retrieved.
+* `awsRegion` - AWS Region used by Transformer to access DynamoDB
+* `dynamodbTable` - DynamoDB table used to store information about duplicate events
+* `purpose` - Always `EVENTS_MANIFEST`
+
+An example of this auxiliary configuration is as follows:
+
+{% highlight json %}
+{
+  "schema": "iglu:com.snowplowanalytics.snowplow.storage/amazon_dynamodb_config/jsonschema/2-0-0",
+  "data": {
+    "name": "eventsManifest",
+    "auth": {
+      "accessKeyId": "fakeAccessKeyId",
+      "secretAccessKey": "fakeSecretAccessKey"
+    },
+    "awsRegion": "us-east-1",
+    "dynamodbTable": "acme-crossbatch-dedupe",
+    "id": "ce6c3ff2-8a05-4b70-bbaa-830c163527da",
+    "purpose": "EVENTS_MANIFEST"
+  }
+}
+{% endhighlight %}
+
+This configuration can be passed to Transformer using the optional `--events-manifest` flag, either as a file path or as a base64-encoded string if the `--base64` flag is also set.
+
+<h2 id="new-config">2. New configuration options</h2>
+
+Version 0.4.0 introduces two new configuration options to the Snowflake Loader config:
+* `maxError` - An optional setting used when writing to Snowflake - a table copy statement will skip an input file when the number of errors in it exceeds the specified number. This can be used to process runs with a certain number of expected bad rows without immediately failing.
+* `jdbcHost` - An optional host for the JDBC driver that has priority over automatically derived hosts.
+
+<h2 id="other-changes">3. Other changes</h2>
+
+This release introduces plenty of other updates:
+
+* The Snowflake Transformer now uses a [custom staging committer][s3committer] which accelerates writing to S3 from Spark by writing task outputs to a temporary directory on the local FS rather than S3. This represents a significant performance improvement by avoiding expensive S3 renaming operations.
```

Comment:
This one definitely deserves its own section, straight after deduplication. It should also include a description of a problem: S3 bottleneck, job was irresponsible to cluster-upscaling, thus incapable for almost anything with more than 1TB per day (rough estimate). And very important to mention that it is not enabled by default.